### PR TITLE
Issue #116: Delete Amenity Booking

### LINKED ISSUE #116: Delete an amenity booking
## ⚠️ Is your feature request related to a problem? Please describe

* Delete an amenity booking

## 💡 Describe the solution you'd like

* `DELETE /amenities/{amenityId}/bookings/{bookingId}`
* Only allow deleting a future booking
* Update postman collection

## Task Relationship

Subtask of: jmprathab/MyHome#94

### PR DESCRIPTION
**Pull request by tomsajuk**
## 🚀 Description

Added endpoint for deleting a booking by booking_id

## 📄 Motivation and Context

To delete a amenity booking

Closes #116 

## 🧪 How Has This Been Tested?

Tested via postman and unit testing

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tomsajuk** [comment]
@mslowiak @jmprathab 
The end point is /booking/{booking_id} as I felt that for deleting a booking only booking id is required. Passing amenity id didn't feel right to me as its not used for deleting.

I changed mapping of AmenityBookingItem to Amenity and User to Many To One. Do suggest if that's okay.

**Comment by mslowiak** [comment]
For me the issue is not clear at this point.
I don't know if the main goal of this issue is to delete a booking where amenityId belongs to or to delete amenity with id from booking where it exists.
@jmprathab you created that story, can you describe your point of view?

File: service/src/main/java/com/myhome/domain/AmenityBookingItem.java [hunk 1]
```diff
@@ -16,20 +16,41 @@
...
 
   @Column(nullable = false, unique = true)
   private String amenityBookingItemId;
-  @OneToOne
+  @ManyToOne(fetch = FetchType.LAZY)
   private Amenity amenity;
-  @Column
+  @Column(nullable = false)
   private LocalDateTime bookingStartDate;
   @Column
```
Comment (jmprathab) [inline service/src/main/java/com/myhome/domain/AmenityBookingItem.java:52]: This needs to be Non-Null as well

**Review by jmprathab — CHANGES_REQUESTED** [review]
@mslowiak 
The goal of this issue is to delete a booking.

@tomsajuk You are right. Amenity ID is not required for deleting a booking. Your approach is well and good.

**Comment by jmprathab** [comment]
@tomsajuk Please look into the following
- Fix conflict
- Please use assertive language in [commit](e9f57722546463f474868480ca119dbb7dfc3242) message
- Please follow the branch name convention as described in [CONTRIBUTING.md](github.com/jmprathab/MyHome/blob/master/CONTRIBUTING.md)

Once these are taken care this is ready to be merged.

**Review by jmprathab — CHANGES_REQUESTED** [review]
@tomsajuk Please look into the following
- Fix conflict
- Use assertive language in [commit](e9f57722546463f474868480ca119dbb7dfc3242) message
- Follow the branch name convention as described in [CONTRIBUTING.md](github.com/jmprathab/MyHome/blob/master/CONTRIBUTING.md)

This PR is ready to be merged after done.

**Comment by tomsajuk** [comment]
Closing this PR as requires branch name change. Please check PR #197

**Pull request closed without merging**